accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,317 @@
+
+When the tool have access to an Uyuni server, then it can run "standalone" or via `mgradm`, allowing to reuse existing Prometheus and Grafana instances.
+
+- The Health Check Tool is not installed or running by default in a Uyuni server.
+- As multiple components (containers) are deployed, resources on a live Uyuni server might be affected.
+- It should be able to reuse existing Prometheus and Grafana instances, allowing integration with existing Monitoring stack.
+
+
+# Security considerations
+In the current design, we are exposing metrics via Prometheus and making them available in Grafana, and more importantly we are exposing logs messages via Loki/LogCLI to CLI and Grafana users. It is important to notice that after running this tool, and until related containers are destroyed, the Grafana Dashboards (and other components like Prometheus and Loki/LogCLI) are exposing metrics and logs messages that may contain sensitive data and information to any non-root user in the system or to anyone that have access to this host in the network.
+
+Promtail pipeline definition can be enhanced to use [replace](https://grafana.com/docs/loki/latest/send-data/promtail/stages/replace/) function, to hide sensitive data from logs. Users must have a configuration file to define the different filters to apply to the Promtail pipeline. Filters must be also configurable via Monitoring Formula.
+
+TLS must be enabled for Promtail, Prometheus and Loki to ensure a secure transport.
+
+Authentication backend for Prometheus and Loki is usually delegated to an authentication reverse proxy. This must be included in the documentation and probably be considered in the Formula.
+
+
+# Learning Curve and Documentation
+System administrators and developers may need to familiarize themselves with Loki/Promtail's log querying language (LogQL) and Grafana's alerting mechanisms. The learning curve associated with these tools could slow down initial adoption and efficiency gains.
+
+In this sense, we must provide a good documentation with tutorials, use cases and examples to help the users to early and smoothly adopt the new monitoring capabilities.
+
+
+# Implementation steps
+
+### Phase 0 (before 5.0 GA):
+
+#### Uyuni Health Check
+- Create RPM packages in OBS/IBS.
+- Use container images building on OBS/IBS for the different components.
+- Provide an "standalone" version only (no integration with Monitoring stack). Via Uyuni server channel (or others so it is available for engineering or supporters).
+- Flag this as a Tech Preview.
+
+#### Saline
+- Research about using TCP instead of IPC sockets for Salt Master internal sockets.
+- Build Saline in OBS/IBS and make it part of the Uyuni Server image
+- Saline runs inside "uyuni-server" container.
```

Comment:
```suggestion
#### Saline

- Research about using TCP instead of IPC sockets for Salt Master internal sockets.
- Build Saline in OBS/IBS and make it part of the Uyuni Server image
- Saline runs inside the `uyuni-server` container.
```